distributed/scheduler.py

```diff
@@ -3854,8 +3854,11 @@ def update_graph(
 
             for a, kv in annotations.items():
                 for k, v in kv.items():
-                    ts = parent._tasks[k]
-                    ts._annotations[a] = v
+                    # Tasks might have been culled, in which case
+                    # we have nothing to annotate.
+                    if k in parent._tasks:
+                        ts = parent._tasks[k]
```

Comment:
Typically we are handling these things with `.get` that avoids 2 retrievals and does justo one. Plus the `is None` check is a pointer comparison in Cython (so quite fast). Pure Python also does well with this kind of code

```suggestion
                    ts = parent._tasks.get(k)
                    if ts is not None:
```